backend/social/queries.py

```diff
@@ -94,23 +95,22 @@ def resolve_get_relationships_between_users(self, info, user_login, user_valued)
         if user_login == user_valued:
             raise ValueError(_('No puedes tener una relación contigo mismo'))
 
+
+
         if user_login.roles.filter(role='OWNER').exists() and user_valued.roles.filter(role='RENTER').exists():
-            properties = Property.objects.filter(owner=user_login)
-            for property in properties:
-                if property.flatmates.filter(id=user_valued.id).exists():
-                    relationships.append('Propietario')
+            properties = Property.objects.filter(owner=user_login).filter(flatmates__in=[user_valued])
+            if properties.exists():
+                relationships.append('Propietario')
 
         if user_login.roles.filter(role='RENTER').exists() and user_valued.roles.filter(role='OWNER').exists():
-            properties = Property.objects.filter(owner=user_valued)
-            for property in properties:
-                if property.flatmates.filter(id=user_login.id).exists():
-                    relationships.append('Inquilino')
+            properties = Property.objects.filter(owner=user_valued).filter(flatmates__in=[user_login])
+            if properties.exists():
+                relationships.append('Inquilino')
 
         if user_login.roles.filter(role='RENTER').exists() and user_valued.roles.filter(role='RENTER').exists():
-            properties = Property.objects.all()
-            for property in properties:
-                if property.flatmates.filter(id=user_login.id).exists() and property.flatmates.filter(id=user_valued.id).exists():
-                    relationships.append('Compañero')
+            properties = Property.objects.filter(flatmates__in=[user_login]).filter(flatmates__in=[user_valued])
```

Comment:
En principio creo que sustituyendo esa línea por `properties = Property.objects.filter(flatmates__in=[user_login, user_valued])` se obtiene el mismo resultado. Prueba a cambiarla y haz pruebas a ver si funciona igual.

Reply:
Esa consulta realiza una operación OR, es decir, te devolvería las propiedades para las que el user_login esta contenido en la lista de usuarios "flatmates " o para las que el user_valued esta contenido en dicha lista. Para que se establezca la relación compañeros es requerido que ambos usuarios estén el mismo piso por lo que sería necesario una operación AND. La manera en la que esta actualmente implementado corresponde con dicha operación AND.

Reply:
Perfecto entonces.